app/updatemanager.go

```diff
@@ -333,6 +337,10 @@ func NewControlMap() *ControlMapPool {
 	}
 }
 
+func (c *ControlMapPool) SetStore(store store.Store) {
```

Comment:
Suggestion:
Since this should always be backed by a DB, can this not be part of the initializer?

Reply:
The primary motivation was to avoid needing this in every test. I can agree that this is perhaps less clean than it could be though, so maybe we should not let test code dictate this. If fixed, this would also remove the need for the `c.store == nil` check that you pointed out in your first comment.

I'll see if I can fix it without too much trouble.